### ISSUE TITLE
Firefox addon is still on version 0.3.7 while Chrome is on 0.5.1

### ISSUE THREAD

**Issue by theolice** [james-fray/tab-reloader#111]
The addon on firefox hasn't been updated in a while and I feel like we're missing out on some improvements.

**Comment by james-fray** [comment]
True, 0.5.x is about moving to manifest v3 and simplifying the interface. 

The service worker-based background is not yet supported on Firefox. Until then, we have to stick to 0.3.x

**Comment by megamorphg** [comment]
Dang, hopefully FF optionally adds that... so I guess that means until then any bugfixes/enhancements will be missed out on? Seems like FF users may need their own fork.

**Comment by madduck** [comment]
Could the new regexp/pattern matching be backported to 0.3.x?

**Comment by james-fray** [comment]
@madduck FF does not support the URLPattern interface yet; https://developer.mozilla.org/en-US/docs/Web/API/URLPattern

**Comment by madduck** [comment]
Regexps would be fine. Not sure why Mozilla had to reinvent the wheel here...

**Comment by megamorphg** [comment]
Looks like it works only with Web Workers
"Note: This feature is available in [Web Workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API)"

I'm still not clear what exactly isn't supported with [service workers](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API) in Firefox. Wondering if someone can break it down for a layman.

Also, when Firefox has manifest v3 does this mean 0.5.x+ will work then?

**Comment by james-fray** [comment]
> when Firefox has manifest v3 does this mean 0.5.x+ will work then?

We may need to wrap a few native methods due to different behavior on FF and Chrome. But it is doable.

**Comment by Omnibrain007** [comment]
https://www.theregister.com/2022/05/20/mozilla_opens_testing_for_manifest/

**Comment by james-fray** [comment]
The latest commit fixes almost all issues to run 0.5.x on FF. Please give it a try and report if you find any issues. Let me know if you need instructions to try on the developer mode.

https://github.com/james-fray/tab-reloader/archive/refs/heads/master.zip -> v3 directory

**Comment by james-fray** [comment]
No need for Nightly
